packages/jotai/src/types.ts

```diff
@@ -0,0 +1,20 @@
+import type { WritableAtom } from 'jotai'
+import type { ReactNode } from 'react'
+
+export interface WithChildren<TValue> {
+  children: (value: TValue) => ReactNode
+}
```

Comment:
Name could be more intuitive.

```suggestion
export interface CallbackChildrenProps<TValue> {
  children: (value: TValue) => ReactNode
}
```